TrafficCapture/trafficReplayer/src/main/java/org/opensearch/migrations/replay/datahandlers/http/NettyJsonBodyConvertHandler.java

```diff
@@ -48,49 +47,92 @@ public static HttpJsonRequestWithFaultingPayload transform(
+            payload.setDisableThrowingPayloadNotLoaded(false);
+        }
+
+        return new ProtectHttpMessageReturnVal(originalPayloadByteBuf, protectedByteBuf);
+    }
+
+    // Undo protectByteBufInHttpMessage after transformations
+    private static void unProtectByteBufInHttpMessage(HttpJsonMessageWithFaultingPayload httpJsonMessage,
+        ProtectHttpMessageReturnVal protectionBufs) throws TransformationException {
+        if (protectionBufs.originalPayloadByteBuf != null) {
             // replace protected byteBuf if was hidden and still there
+            // no risk of PayloadNotLoadedException since originalPayloadByteBuf already retrieved
             var transformedInlinedBinary = httpJsonMessage.payload().get(JsonKeysForHttpMessage.INLINED_BINARY_BODY_DOCUMENT_KEY);
-            assert protectedByteBuf != null : "Expected protectedByteBuf to be defined if innerPayloadByteBuf is";
-            if (protectedByteBuf.equals(transformedInlinedBinary)) {
-                innerPayloadByteBuf.readerIndex(protectedByteBuf.readerIndex());
-                innerPayloadByteBuf.writerIndex(protectedByteBuf.writerIndex());
-                httpJsonMessage.payload().put(JsonKeysForHttpMessage.INLINED_BINARY_BODY_DOCUMENT_KEY, innerPayloadByteBuf);
+            assert protectionBufs.protectedPayloadByteBuf != null : "Expected protectedByteBuf to be defined if innerPayloadByteBuf is";
+            // Shallow check to see if object is the same or replaced
+            if (protectionBufs.protectedPayloadByteBuf == transformedInlinedBinary) {
+                protectionBufs.originalPayloadByteBuf.readerIndex(protectionBufs.protectedPayloadByteBuf.readerIndex());
+                protectionBufs.originalPayloadByteBuf.writerIndex(protectionBufs.protectedPayloadByteBuf.writerIndex());
+                httpJsonMessage.payload().put(JsonKeysForHttpMessage.INLINED_BINARY_BODY_DOCUMENT_KEY, protectionBufs.originalPayloadByteBuf);
             } else {
-                innerPayloadByteBuf.release();
+                // Transformation changed byteBuf, no longer needing originalPayloadByteBuf
+                protectionBufs.originalPayloadByteBuf.release();
                 if (transformedInlinedBinary == null) {
                     httpJsonMessage.payload().remove(JsonKeysForHttpMessage.INLINED_BINARY_BODY_DOCUMENT_KEY);
-                } else if (!(transformedInlinedBinary instanceof ByteBuf)) {
-                    throw new UnsupportedOperationException("Type of " + JsonKeysForHttpMessage.INLINED_BINARY_BODY_DOCUMENT_KEY
-                        + " not supported.");
+                } else if (transformedInlinedBinary instanceof ByteBuf) {
+                    // Transformation set new ByteBuf, verify it's valid
+                    var transformedInlinedBinaryByteBuf = (ByteBuf) transformedInlinedBinary;
+                    // Touch for easier debugging if later leak detected
+                    // e.g. if received unreleasable ByteBuf
```

Comment:
Playing w/ netty & even looking at touch backtraces for years & this comment makes me understand when I would want to use touch().  Thanks!